azurerm/internal/services/eventhub/eventhub_namespace_dedicated_resource.go

```diff
@@ -0,0 +1,415 @@
+package eventhub
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/eventhub/mgmt/2018-01-01-preview/eventhub"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+// Default Authorization Rule/Policy created by Azure, used to populate the
+// default connection strings and keys
+var eventHubNamespaceDedicatedResourceName = "azurerm_eventhub_namespace_dedicated"
+
+func resourceArmEventHubNamespaceDedicated() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmeventHubNamespaceDedicatedCreateUpdate,
+		Read:   resourceArmeventHubNamespaceDedicatedRead,
+		Update: resourceArmeventHubNamespaceDedicatedCreateUpdate,
+		Delete: resourceArmeventHubNamespaceDedicatedDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateEventHubNamespaceName(),
+			},
+
+			"cluster_id": {
```

Comment:
since this is kind of ambiguous, it'd be worth probably calling this out as:

```suggestion
			"dedicated_cluster_id": {
```

(FWIW chatting internally we believe the `azurerm_eventhub_cluster` resource should probably be named `azurerm_eventhub_dedicated_cluster`, but that's a larger discussion)

Reply:
If we're going to ditch dedicated resources, this change will become obsolete then? I'm getting lost in diverging review opinions to be honest :)